src/Neo/Network/P2P/Payloads/Conflicts.cs

```diff
@@ -0,0 +1,47 @@
+using Neo.IO;
+using Neo.Json;
+using Neo.Persistence;
+using Neo.SmartContract.Native;
+using System.IO;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class Conflicts : TransactionAttribute
+    {
+        /// <summary>
+        /// Indicates the conflict transaction hash.
+        /// </summary>
+        public UInt256 Hash;
+
+        public override TransactionAttributeType Type => TransactionAttributeType.Conflicts;
+
+        public override bool AllowMultiple => true;
+
+        public override int Size => base.Size + Hash.Size;
+
+        protected override void DeserializeWithoutType(ref MemoryReader reader)
+        {
+            Hash = reader.ReadSerializable<UInt256>();
+        }
+
+        protected override void SerializeWithoutType(BinaryWriter writer)
+        {
+            writer.Write(Hash);
+        }
+
+        public override JObject ToJson()
+        {
+            JObject json = base.ToJson();
+            json["hash"] = Hash.ToString();
+            return json;
+        }
+
+        public override bool Verify(DataCache snapshot, Transaction tx)
+        {
```

Comment:
Ensure that the the hash is not duplicated in another Conflict attribute?

Reply:
I don't think it's needed, because all Conflicts attributes are paid and it doesn't affect the resulting storage state. If the transaction creator wants to pay more for the duplicating attributes - it's his right.

HashSet is implemented in the mempool, it's a nice optimisation.